core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1086,14 +1086,14 @@ private void convertWhere(
   private void replaceSubQueries(
       final Blackboard bb,
       final SqlNode expr,
-      RelOptUtil.Logic logic) {
+      RelOptUtil.Logic logic, boolean isWhere) {
```

Comment:
Do you think it makes sense to add a small `enum` instead of `boolean` here?

It would make the usages easier to understand.
So far almost all the usages pass `isWhere=false`, so does it make sense to add an overloaded method for that?

Reply:
you are right, let me try it

Reply:
I have modified